pkg/apis/operator/v1alpha1/type.go

```diff
@@ -0,0 +1,886 @@
+	// +optional
+	Images *Images `json:"images,omitempty"`
+
+	// Networking holds configuration for the networking topology of the cluster.
+	// +optional
+	Networking *Networking `json:"networking,omitempty"`
+
+	// Cert define the certificate details for karmada control plane.
+	// TODO: whether need to be exposed to the user?
+	// +optional
+	// Cert *CertConfig `json:"cert,omitempty"`
+
+	// Etcd holds configuration for etcd.
+	// +required
+	Etcd *Etcd `json:"etcd,omitempty"`
+
+	// APIServer contains extra settings for the API server control plane component
+	// +optional
+	APIServers *APIServerComponents `json:"apiServers,omitempty"`
+
+	// ControllerManager contains extra settings for the controller manager control plane component
+	// +optional
+	ControllerManagers *ControllerManagerComponents `json:"controllerManagers,omitempty"`
+
+	// Scheduler contains extra settings for the scheduler control plane component
+	// +optional
+	Schedulers *SchedulerComponents `json:"schedulers,omitempty"`
+
+	// Webhooks contains extra settings for the webhook component
+	// +optional
+	Webhooks *WebhookComponents `json:"webhooks,omitempty"`
+
+	// FeatureGates enabled by the user.
+	// - Failover: https://karmada.io/docs/userguide/failover/#failover
+	// - GragscefulEviction: https://karmada.io/docs/userguide/failover/#graceful-eviction-feature
+	// - PropagateDeps: https://karmada.io/docs/userguide/scheduling/propagate-dependencies
+	// - CustomizedClusterResourceModeling: https://karmada.io/docs/userguide/scheduling/cluster-resources#start-to-use-cluster-resource-models
+	// More info: https://github.com/karmada-io/karmada/blob/master/pkg/features/features.go
+	// +optional
+	FeatureGates map[string]bool `json:"featureGates,omitempty"`
+}
+
+// Networking contains elements describing cluster's networking configuration
+type Networking struct {
+	// ServiceSubnet is the subnet used by k8s services. Defaults to "10.96.0.0/12".
+	// +optional
+	ServiceSubnet string `json:"serviceSubnet,omitempty"`
+
+	// DNSDomain is the dns domain used by k8s services. Defaults to "cluster.local".
+	// +optional
+	DNSDomain string `json:"dnsDomain,omitempty"`
+}
+
+// Etcd contains elements describing Etcd configuration.
+type Etcd struct {
+	// Local provides configuration knobs for configuring the built-in etcd instance
+	// Local and External are mutually exclusive
+	// +optional
+	Local *LocalEtcd `json:"local,omitempty"`
+
+	// External describes how to connect to an external etcd cluster
+	// Local and External are mutually exclusive
+	// +optional
+	External *ExternalEtcd `json:"external,omitempty"`
+}
+
+// Etcd contains elements describing karmada control plane configuration.
+type CertConfig struct {
+	// PeerCertSANs sets extra Subject Alternative Names for the karmada control plane signing cert.
+	// +optional
+	CertSANs []string `json:"certSANs,omitempty"`
+
+	// Expiry sets certificate expiration time, it's 5 years by default.
```

Comment:
Generally, in Kubernetes, the the Expiration of CA cert is 10 years and the client cert is 1 year by default.